OsmAnd/src/net/osmand/plus/MapMarkersHelper.java

```diff
@@ -439,6 +397,7 @@ public void syncGroup(MarkersSyncGroup group) {
 
 			List<WptPt> gpxPoints = new LinkedList<>(gpx.getPoints());
 			for (WptPt pt : gpxPoints) {
+				group.setColor(pt.getColor(ContextCompat.getColor(ctx, R.color.marker_red)));
```

Comment:
ContextCompat.getColor(ctx, R.color.marker_red)) in loop

Reply:
Color may differ for each point, which means it is necessary.